wpilibcExamples/src/main/cpp/examples/I2CCommunication/cpp/Robot.cpp

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#include <frc/DriverStation.h>
+#include <frc/I2C.h>
+#include <frc/TimedRobot.h>
+#include <fmt/format.h>
+#include <wpi/SmallString.h>
+#include <wpi/StringRef.h>
+#include <wpi/raw_ostream.h>
+
+/**
+ * This is a sample program demonstrating how to communicate to a light
+ * controller from the robot code using the roboRIO's I2C port.
+ */
+class Robot : public frc::TimedRobot {
+ public:
+  void RobotPeriodic() override {
+    // Creates a string to hold current robot state information, including
+    // alliance, enabled state, operation mode, and match time. The message
+    // is sent in format "AEM###" where A is the alliance color, (R)ed or
+    // (B)lue, E is the enabled state, (E)nabled or (D)isabled, M is the
+    // operation mode, (A)utonomous or (T)eleop, and ### is the zero-padded
+    // time remaining in the match.
+    //
+    // For example, "RET043" would indicate that the robot is on the red
+    // alliance, enabled in teleop mode, with 43 seconds left in the match.
+    wpi::SmallString<128> data;
+    wpi::raw_svector_ostream os(data);
+    os << (frc::DriverStation::GetAlliance() == frc::DriverStation::Alliance::kRed ? "R" : "B")
+       << (frc::DriverStation::IsEnabled() ? "E" : "D") << (frc::DriverStation::IsAutonomous() ? "A" : "T")
+       << fmt::format("%03d", frc::DriverStation::GetMatchTime());
```

Comment:
fmtlib doesn't use printf-style format strings.

Worth replacing the ostream usage with just fmtlib though, perhaps?